### PR TITLE
[AC-1708] Teams Starter Plan

### PR DESCRIPTION
**Pull request by cturnbull-bitwarden**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Relies on https://github.com/bitwarden/server/pull/3320 to be merged first

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/Models/StaticStore/Plans/Families2019Plan.cs [hunk 1]
```diff
@@ -30,9 +30,9 @@ private record Families2019PasswordManagerFeatures : PasswordManagerPlanFeatures
     {
         public Families2019PasswordManagerFeatures()
         {
-            BaseSeats = 5;
+            BaseSeats = 6;
             BaseStorageGb = 1;
-            MaxSeats = 5;
+            MaxSeats = 6;
```
[the remark this answers is not in the document]
Comment (cturnbull-bitwarden) [reply src/Core/Models/StaticStore/Plans/Families2019Plan.cs:35]: I did! This is from Priya, who wants to make sure that all 2019 plans have feature parity with current-year plans

File: src/Api/Controllers/PlansController.cs [hunk 2]
```diff
@@ -33,9 +33,12 @@ public class PlansController : Controller
...
         var plansUpgradeIsEnabled = _featureService.IsEnabled(FeatureFlagKeys.BillingPlansUpgrade, _currentContext);
-        var data = StaticStore.Plans;
-        var responses = data
-            .Where(plan => plansUpgradeIsEnabled || plan.Type <= PlanType.EnterpriseAnnually2020)
+        var teamsStarterPlanIsEnabled = _featureService.IsEnabled(FeatureFlagKeys.BillingStarterPlan, _currentContext);
+        var responses = StaticStore.Plans
+            .Where(plan => (plansUpgradeIsEnabled ||
+                            plan.Type <= PlanType.EnterpriseAnnually2020 ||
+                            plan.Type == PlanType.TeamsStarter) &&
+                           (teamsStarterPlanIsEnabled || plan.Product != ProductType.TeamsStarter))
```
[the remark this answers is not in the document]
Comment (cturnbull-bitwarden) [reply src/Api/Controllers/PlansController.cs]: Just pushed a change splitting the logic a bit, and added some comments explaining it. Let me know if it's still not clear

File: src/Core/Models/StaticStore/Plans/TeamsStarterPlan.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+        HasGroups = true;
+        HasDirectory = true;
+        HasEvents = true;
+        HasTotp = true;
+        Has2fa = true;
+        HasApi = true;
+        UsersGetPremium = true;
+
+        UpgradeSortOrder = 2;
```
Comment (cyprain-okeke) [inline src/Core/Models/StaticStore/Plans/TeamsStarterPlan.cs:26]: Are customer able to upgrade from TeamsStarterPlan to Enterprise

Comment (cturnbull-bitwarden) [reply src/Core/Models/StaticStore/Plans/TeamsStarterPlan.cs:26]: No, not yet at least. That will be included in the December release

**Review by amorask-bitwarden — APPROVED** [review]
👍 Nice work!

**Review by cyprain-okeke — APPROVED** [review]

**Review by Thomas-Avery — APPROVED** [review]
SM changes look good.

**Review by vincentsalucci — APPROVED** [review]

**Review by amorask-bitwarden — APPROVED** [review]

**Review by cyprain-okeke — APPROVED** [review]

**Review by vincentsalucci — APPROVED** [review]

**Review by coltonhurst — APPROVED** [review]

**Pull request merged**